File: plane-reporter-neo4j/src/main/java/com/vijayepa/aircraft/v2/polling/PlaneFinderPoller.java
package com.vijayepa.aircraft.v2.polling;

import com.vijayepa.aircraft.v2.cacheacccess.AircraftRepository;
import com.vijayepa.aircraft.v2.webaccess.AircraftClient;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@EnableScheduling
public class PlaneFinderPoller {

    private final AircraftRepository aircraftRepository;

    private final AircraftClient aircraftClient;


    public PlaneFinderPoller(
            AircraftRepository aircraftRepository,
            AircraftClient aircraftClient) {
        this.aircraftRepository = aircraftRepository;
        this.aircraftClient = aircraftClient;
    }

    @Scheduled(fixedRate = 1000)
    private void pollPlanes() {
//        aircraftRepository.deleteAll();

//        aircraftClient.getAircrafts()
//                .forEach(aircraftRepository::save);

        aircraftRepository.findAll()
                .forEach(System.out::println);
    }
}

File: plane-reporter-mongo/src/main/kotlin/com/vijayvepa/planereportermongo/poller/PlaneFinderPoller.kt
package com.vijayvepa.planereportermongo.poller

import com.vijayvepa.planereportermongo.dataaccess.AircraftRepository
import com.vijayvepa.planereportermongo.model.Aircraft
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient

@Component
@EnableScheduling
class PlaneFinderPoller(private val repository: AircraftRepository) {
    private val client = WebClient.create("http://localhost:7634/aircraft")

    @Scheduled(fixedRate = 1000)
    private fun pollPlanes() {
        repository.deleteAll();
        client.get().retrieve().bodyToFlux(Aircraft::class.java)
            .filter { !it.reg.isNullOrEmpty() }
            .toStream().forEach { repository.save(it) }

        println("-- All Aircraft")
        repository.findAll().forEach { println(it) }
    }
}